javascript/ql/src/semmle/javascript/Stmt.qll

```diff
@@ -818,6 +818,15 @@ abstract class EnhancedForLoop extends LoopStmt {
     result = getIterator().(DeclStmt).getADecl()
   }
 
+  /**
+   * Gets the property, variable, or destructuring pattern occurring as the iterator
+   * expression in this `for`-`in` or `for`-`of` loop.
+   */
+  Expr getLValue() {
+    result = getIteratorExpr() or
```

Comment:
It looks like this will also return the variable declarator (if there is one), since variable declarators are expressions (for no particularly good reason). Is that intended?

Reply:
Good point, I've restricted the result